pkg/audit/workflow_auditor.go

```diff
@@ -0,0 +1,410 @@
+// Package audit provides audit logging functionality for ToolHive.
+package audit
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/stacklok/toolhive/pkg/auth"
+)
+
+// WorkflowAuditor provides audit logging for workflow execution.
+// This interface abstracts workflow-specific audit operations from the
+// HTTP middleware-based Auditor.
+type WorkflowAuditor struct {
+	auditLogger *slog.Logger
+	config      *Config
+	component   string
+}
+
+// NewWorkflowAuditor creates a new workflow auditor.
+// If config is nil, creates a default configuration with stdout logging.
+func NewWorkflowAuditor(config *Config) (*WorkflowAuditor, error) {
+	if config == nil {
+		config = DefaultConfig()
+	}
+
+	logWriter, err := config.GetLogWriter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create log writer: %w", err)
+	}
+
+	return &WorkflowAuditor{
+		auditLogger: NewAuditLogger(logWriter),
+		config:      config,
+		component:   "vmcp-composer",
+	}, nil
+}
+
+// LogWorkflowStarted logs the start of workflow execution.
+func (w *WorkflowAuditor) LogWorkflowStarted(
+	ctx context.Context,
+	workflowID string,
+	workflowName string,
+	parameters map[string]any,
+	timeout time.Duration,
+) {
+	if !w.config.ShouldAuditEvent(EventTypeWorkflowStarted) {
+		return
+	}
+
+	source := w.extractSource(ctx)
+	subjects := w.extractSubjects(ctx)
+
+	event := NewAuditEvent(
+		EventTypeWorkflowStarted,
+		source,
+		OutcomeSuccess,
+		subjects,
+		w.component,
+	)
+
+	target := map[string]string{
+		TargetKeyWorkflowID:   workflowID,
+		TargetKeyWorkflowName: workflowName,
+		TargetKeyType:         TargetTypeWorkflow,
+	}
+	event.WithTarget(target)
+
+	// Add workflow parameters as data (if configured)
+	if w.config.IncludeRequestData {
+		data := map[string]any{
```

Comment:
Quick question - the HTTP auditor wraps its data in `{"request": ..., "response": ...}` structure, while this uses workflow-specific fields. Totally makes sense since workflows aren't really request/response, but curious if you considered matching the structure for easier log parsing?

No strong opinion, just wondering if there was a specific reason to go with the different approach.